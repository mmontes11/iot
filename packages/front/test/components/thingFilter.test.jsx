import React from "react";
import ThingFilter from "components/thingFilter";
import { shallow } from "enzyme";

describe("components/thingFilter", () => {
  it("renders ThingFilter in initial state", () => {
    const thingFilter = {
      label: "Select filter:",
      items: ["foo", "bar"],
      isActive: false,
      isLoading: false,
      isDisabled: false,
      onButtonClick: () => undefined,
      onItemClick: () => undefined,
      onDelete: () => undefined,
    };
    const wrapper = shallow(<ThingFilter thingFilter={thingFilter} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("simulates a click in delete button", () => {
    const onDelete = jest.fn();
    const thingFilter = {
      label: "Select filter:",
      items: ["foo", "bar"],
      isActive: false,
      isLoading: false,
      isDisabled: false,
      onButtonClick: () => undefined,
      onItemClick: () => undefined,
      onDelete,
    };
    const wrapper = shallow(<ThingFilter thingFilter={thingFilter} />);
    wrapper.find("#delete-button").simulate("click");
    expect(onDelete).toHaveBeenCalled();
  });
});
