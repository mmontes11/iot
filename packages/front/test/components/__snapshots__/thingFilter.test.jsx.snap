// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/thingFilter renders ThingFilter in initial state 1`] = `
<div
  className="box"
>
  <div
    className="columns"
  >
    <div
      className="column is-10"
    >
      <Dropdown
        buttonStyle=""
        iconStyle="fa-angle-down"
        isActive={false}
        isDisabled={false}
        isLoading={false}
        items={
          Array [
            "foo",
            "bar",
          ]
        }
        label="Select filter:"
        onButtonClick={[Function]}
        onDelete={[Function]}
        onItemClick={[Function]}
      />
    </div>
    <div
      className="column is-center"
    >
      <button
        className="delete is-medium"
        id="delete-button"
        onClick={[Function]}
        tabIndex={0}
      >
        <FormattedMessage
          id="Delete"
          values={Object {}}
        />
      </button>
    </div>
  </div>
</div>
`;
